File: server/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/ManagerStatusUpdateRequest.java
package edu.cnm.deepdive.seesomethingabq.model.dto;

/**
 * Request body DTO for updating manager status of a user.
 */
public class ManagerStatusUpdateRequest {

  private boolean manager;

  /**
   * Returns whether the user should have manager privileges.
   *
   * @return {@code true} if manager; {@code false} otherwise.
   */
  public boolean isManager() {
    return manager;
  }

  /**
   * Sets whether the user should have manager privileges.
   *
   * @param manager {@code true} if manager; {@code false} otherwise.
   */
  public void setManager(boolean manager) {
    this.manager = manager;
  }

}

File: app/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/ManagerStatusUpdateRequest.kt
package edu.cnm.deepdive.seesomethingabq.model.dto

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Request body for PATCH /manager/users/{externalId}/manager-status.
 *
 * Server expects a JSON object with a single boolean property named "manager".
 */
data class ManagerStatusUpdateRequest(
  @Expose
  @SerializedName("manager")
  val manager: Boolean,
)

